src/Symfony/Component/Validator/Constraints/ChoiceValidator.php

```diff
@@ -55,6 +59,13 @@ public function validate($value, Constraint $constraint)
                 throw new ConstraintDefinitionException('The Choice constraint expects a valid callback.');
             }
             $choices = $choices();
+        } elseif (is_a($constraint->choices, \BackedEnum::class, true)) {
+            $choices = array_map(
+                function (\BackedEnum $enum) {
+                    return $enum->value;
+                },
+                $constraint->choices::cases()
```

Comment:
this might fail when choices is a string, isn't it?
does this mean there might be a missing test case?

Reply:
when reaching this, `$constraint->choices` is a string containing a class name, which is then working to call the static method.